src/Runner.js

```diff
@@ -94,77 +86,56 @@ export default class Runner {
   }
 
   onStatusUpdate = (statusEvent: StatusEvent) => {
-    let running = this.running;
-
-    if (!running) {
+    if (!this.running) {
       return Logger.info(
         statusEvent,
         'No build running, status event is irrelevant'
       );
     }
 
+    let running = this.running;
+
     if (running.buildId !== statusEvent.buildId) {
       return Logger.warn(
-        {
-          statusEvent,
-          running
-        },
+        { statusEvent, running },
         `StatusEvent buildId doesn't match currently running buildId – ${
           statusEvent.buildId
         } !== ${running.buildId || ''}`
       );
     }
 
-    Logger.info(
-      {
-        statusEvent,
-        running
-      },
-      'Build status update'
-    );
+    Logger.info({ statusEvent, running }, 'Build status update');
 
     this.running = running = {
       ...running,
       buildStatus: statusEvent.buildStatus
     };
 
+    let addToHistory = () =>
+      this.history.set(statusEvent, { ...running, finishedTime: new Date() });
+
     if (statusEvent.passed) {
       this.mergePassedBuild(running);
+      addToHistory();
       this.running = null;
       this.next();
     } else if (statusEvent.failed) {
-      Logger.error(
-        {
-          running,
-          statusEvent
-        },
-        'Land build failed'
-      );
+      Logger.error({ running, statusEvent }, 'Land build failed');
+      addToHistory();
       this.running = null;
       this.next();
     } else if (statusEvent.stopped) {
-      Logger.warn(
-        {
-          running,
-          statusEvent
-        },
-        'Land build has been stopped'
-      );
+      Logger.warn({ running, statusEvent }, 'Land build has been stopped');
+      addToHistory();
       this.running = null;
       this.next();
     }
   };
 
   mergePassedBuild(running: LandRequest) {
     const pullRequestId = running.pullRequestId;
-    Logger.info(
-      {
-        pullRequestId,
-        running
-      },
-      'Merging pull request'
-    );
-    this.client.mergePullRequest(pullRequestId);
+    Logger.info({ pullRequestId, running }, 'Merging pull request');
+    // this.client.mergePullRequest(pullRequestId);
```

Comment:
I'll uncomment this once its all merged in ;P

Reply:
🤦‍♂️